src/Facility.CodeGen.CSharp/CSharpGenerator.cs

```diff
@@ -1228,6 +1232,7 @@ private void GenerateFieldProperties(CodeWriter code, IEnumerable<ServiceFieldIn
 				code.WriteLine($"[Newtonsoft.Json.JsonProperty(\"{fieldInfo.Name}\")]");
 				code.WriteLine($"[System.Text.Json.Serialization.JsonPropertyName(\"{fieldInfo.Name}\")]");
 			}
+			code.WriteLine(FormattableString.Invariant($"[ProtoMember({++protoMember})]"));
```

Comment:
If we made this official, we'd probably want to support an FSD attribute, e.g. `[proto(id: 3)]`. Auto IDs would be nice (and maybe even worth supporting) but dangerous.

Reply:
Definitely. I'd want to write a tool to automatically add the attributes to existing FSDs.